### PR TITLE
Removed partial installplan condition checks and added csv check in operand reconciliation

### PR DESCRIPTION
**Pull request by YCShen1010**
Signed-off-by: YCShen1010 
Related issue: [54892](https://github.ibm.com/IBMPrivateCloud/roadmap/issues/54892)

### CODE REVIEW

**Comment by YCShen1010** [comment]
Code is still in testing according to the  [test plan](https://github.ibm.com/IBMPrivateCloud/roadmap/issues/54892#issuecomment-48327093) 

Test Case 1: fresh install `ibm-mongodb-operator` as example and pieces of logs from odlm pod:
```
255 | W0922 17:40:23.151958 1 manager.go:348] The ClusterServiceVersion for Subscription ibm-mongodb-operator is not ready. Will check it again
256 | W0922 17:40:23.151986 1 reconcile_operand.go:171] ClusterServiceVersion for the Subscription ibm-mongodb-operator in the namespace ibm-common-services is not ready yet, retry
268 | E0922 17:40:43.229915 1 reconcile_operand.go:178] ClusterServiceVersion ibm-mongodb-operator.v1.15.0 status is not succeeded in the namespace ibm-common-services
278 | E0922 17:40:46.265156 1 reconcile_operand.go:178] ClusterServiceVersion ibm-mongodb-operator.v1.15.0 status is not succeeded in the namespace ibm-common-services
316 | I0922 17:41:03.519246 1 operandrequest_controller.go:115] Reconciling OperandRequest: ibm-common-services/ibm-mongodb-request
318 | I0922 17:41:03.557508 1 operandrequest_controller.go:115] Reconciling OperandRequest: ibm-common-services/ibm-mongodb-request
321 | I0922 17:41:03.597234 1 reconcile_operator.go:48] Reconciling Operators for OperandRequest: ibm-common-services/ibm-mongodb-request
323 | I0922 17:41:03.633549 1 reconcile_operator.go:136] Finished reconciling Operators for OperandRequest: ibm-common-services/ibm-mongodb-request
324 | I0922 17:41:03.633578 1 reconcile_operand.go:50] Reconciling Operands for OperandRequest: ibm-common-services/ibm-mongodb-request
326 | I0922 17:41:03.646601 1 reconcile_operand.go:221] Finished reconciling Operands for OperandRequest: ibm-common-services/ibm-mongodb-request
```

File: controllers/operandrequest/reconcile_operand.go [hunk 1]
```diff
@@ -170,17 +155,31 @@ func (r *Reconciler) reconcileOperand(ctx context.Context, requestInstance *oper
...
+			if csv == nil {
+				klog.Warningf("ClusterServiceVersion for the Subscription %s in the namespace %s is not ready yet, retry", operatorName, namespace)
+				requestInstance.SetMemberStatus(operand.Name, operatorv1alpha1.OperatorInstalling, "", &r.Mutex)
+				continue
+			}
+			
+			//Check csv status
+			if csv.Status.Phase != olmv1alpha1.CSVPhaseSucceeded  {
+				klog.Errorf("ClusterServiceVersion %s status is not succeeded in the namespace %s", csv.Name, csv.Namespace)
+				requestInstance.SetMemberStatus(operand.Name, operatorv1alpha1.OperatorFailed, "", &r.Mutex)
```
Comment (Daniel-Fan) [inline controllers/operandrequest/reconcile_operand.go:179]: Could we used `operatorv1alpha1.OperatorInstalling` if the `Phase` is not in `Succeeded`

File: controllers/operandrequest/reconcile_operand.go [hunk 2]
```diff
@@ -170,17 +155,31 @@ func (r *Reconciler) reconcileOperand(ctx context.Context, requestInstance *oper
...
+			}
+
+			if csv == nil {
+				klog.Warningf("ClusterServiceVersion for the Subscription %s in the namespace %s is not ready yet, retry", operatorName, namespace)
+				requestInstance.SetMemberStatus(operand.Name, operatorv1alpha1.OperatorInstalling, "", &r.Mutex)
+				continue
+			}
+			
+			//Check csv status
+			if csv.Status.Phase != olmv1alpha1.CSVPhaseSucceeded  {
```
Comment (Daniel-Fan) [inline controllers/operandrequest/reconcile_operand.go:177]: Could we combine those two conditions check together by using `else if` for the second one.
```
if csv == nil {}
else if csv.Status.Phase != olmv1alpha1.CSVPhaseSucceeded {}
```

File: controllers/operandrequest/reconcile_operand.go [hunk 3]
```diff
@@ -170,17 +155,31 @@ func (r *Reconciler) reconcileOperand(ctx context.Context, requestInstance *oper
...
+				klog.Errorf("ClusterServiceVersion %s status is not succeeded in the namespace %s", csv.Name, csv.Namespace)
```
Comment (Daniel-Fan) [inline controllers/operandrequest/reconcile_operand.go:178]: Could we use the `Warning` msg here with the information
```
ClusterServiceVersion %s phase is not succeeded in the namespace %s yet, retry
```

**Comment by YCShen1010** [comment]
Test Case II: upgrade manually to reproduce issue

- new installplan is generated for new version of operators
```
install-hhwh6   ibm-common-service-operator.v3.19.4    Manual     false
```
- logs from odlm pod
```
I0923 03:19:48.385691 1 reconcile_operand.go:157] Installplan ibm-common-services/install-hhwh6 is waiting for approval
```
- can not propagate sizing profile updates to `cert-manager` CR

memory size in `cert-manager` CR compared with value in `operandConfig` (large):
```
  certManagerController:
    resources:
      limits:
        cpu: 80m
        memory: 530Mi
      requests:
        cpu: 20m
        memory: 230Mi
  certManagerCAInjector:
    resources:
      limits:
        cpu: 100m
        memory: 520Mi
      requests:
        cpu: 20m
        memory: 410Mi
```
```
certManagerController:
            resources:
              limits:
                cpu: 80m
                memory: 1010Mi
              requests:
                cpu: 20m
                memory: 510Mi
certManagerCAInjector:
            resources:
              limits:
                cpu: 100m
                memory: 1000Mi
              requests:
                cpu: 30m
                memory: 500Mi
          
```

**Review by Daniel-Fan — APPROVED** [review]
/lgtm

**Pull request merged**